tribits/core/package_arch/TribitsExternalPackageFindTplHelpers.cmake

```diff
@@ -0,0 +1,132 @@
+# ``find_package(<externalPkg>)`` for external package that uses modern CMake
+# IMPORTED targets.
+#
+# Usage::
+#
+#   tribits_external_package_create_imported_all_libs_target_and_config_file(
+#     <tplName>
+#     INNER_FIND_PACKAGE_NAME <externalPkg>
+#     IMPORTED_TARGETS_FOR_ALL_LIBS <importedTarget0> <importedTarget1> ... )
+#
+# This function is called in a TriBITS ``FindTPL<tplName>.cmake`` wrapper
+# module after it calls ``find_package(<externalPkg>)`` and then creates the
+# IMPORTED target ``<tplName>::all_libs`` from the list of IMPORTED targets
+# ``<importedTarget0> <importedTarget1> ...`` which are defined from the call
+# ``find_package(<externalPkg>)``.  This function also takes care of
+# generating the correct ``<tplName>Config.cmake`` file under the directory::
+#
+#   ${${PROJECT_NAME}_BINARY_DIR}/${${PROJECT_NAME}_BUILD_DIR_EXTERNAL_PKGS_DIR}
+#
+# The generated ``<tplName>Config.cmake`` file calls
+# ``find_dependency(<externalPkg>)`` (with no other argument) and then, again,
+# defines the correct imported library dependency.
+#
+# For more details, see `Creating FindTPL*.cmake using find_package() with
+# IMPORTED targets`_.
+#
+function(tribits_external_package_create_imported_all_libs_target_and_config_file
+    tplName
+  )
+
+  # Parse arguments
+  cmake_parse_arguments(
+     PARSE_ARGV 1
+     PARSE "" "" # prefix, options, one_value_keywords
+     "INNER_FIND_PACKAGE_NAME;IMPORTED_TARGETS_FOR_ALL_LIBS"  #multi_value_keywords
+     )
+  tribits_check_for_unparsed_arguments(PARSE)
+  tribits_assert_parse_arg_one_value(PARSE  INNER_FIND_PACKAGE_NAME)
+  tribits_assert_parse_arg_one_or_more_values(PARSE IMPORTED_TARGETS_FOR_ALL_LIBS)
+
+  # Create imported target <tplName>::all_libs
+  add_library(${tplName}::all_libs  INTERFACE  IMPORTED  GLOBAL)
+  foreach (importedTarget  IN LISTS  PARSE_IMPORTED_TARGETS_FOR_ALL_LIBS)
+    target_link_libraries(${tplName}::all_libs  INTERFACE  ${importedTarget})
+  endforeach()
+
+  # Create <tplName>Config.cmake file
+  tribits_external_package_create_package_config_file_with_imported_targets(
+    ${tplName}
+    INNER_FIND_PACKAGE_NAME ${PARSE_INNER_FIND_PACKAGE_NAME}
+    IMPORTED_TARGETS_FOR_ALL_LIBS ${PARSE_IMPORTED_TARGETS_FOR_ALL_LIBS} )
+
+endfunction()
+
+
+function(tribits_external_package_create_package_config_file_with_imported_targets
+    tplName
+  )
+
+  # Parse arguments
+  cmake_parse_arguments(
+     PARSE_ARGV 1
+     PARSE "" "" # prefix, options, one_value_keywords
+     "INNER_FIND_PACKAGE_NAME;IMPORTED_TARGETS_FOR_ALL_LIBS"  #multi_value_keywords
+     )
+  tribits_check_for_unparsed_arguments(PARSE)
+  tribits_assert_parse_arg_one_value(PARSE  INNER_FIND_PACKAGE_NAME)
+  tribits_assert_parse_arg_one_or_more_values(PARSE IMPORTED_TARGETS_FOR_ALL_LIBS)
+  set(externalPkg ${PARSE_INNER_FIND_PACKAGE_NAME})
+
+  # Create <tplName>Config.cmake file
+  set(configFileStr "")
+  string(APPEND configFileStr
```

Comment:
**Minor:** It seems a bit strange to me to set a string to empty and then immediately append to it, but I guess it's a little more consistent.

Reply:
Yea, after years of experience I have found the idiom of "initialize to empty then append"  to be more robust under future refactorings and extensions.